ruff_lsp/server.py

```diff
@@ -1137,22 +1563,28 @@ def _executable_version(executable: str) -> Version:
     return EXECUTABLE_VERSIONS[executable].version
 
 
-async def _run_check_on_document(
-    document: workspace.TextDocument,
+class DocumentSource(NamedTuple):
+    """The source of a document."""
+
+    path: str
+    """The path to the document."""
+
+    text: str
+    """The text of the document."""
```

Comment:
This is so that we can pass a different source to Ruff than the one in the document. For example, in `Format Cell` document, we'll pass the notebook representation containing a single cell and then compare against the original source. Refer to `format_document` function.